### PR TITLE
Remove ignored typing error of import of `CachingProtocolMeta`

### PR DESCRIPTION
**Pull request by flaeppe**
Closes: #243

Refs:
- https://github.com/posita/numerary/pull/17

### CODE REVIEW

**Comment by antonagestam** [comment]
Don't know what's up with GA, I'll try closing and re-opening the PR.

File: src/phantom/_utils/types.py [hunk 1]
```diff
@@ -1,6 +1,10 @@
+import sys
 from typing import TypeVar
 
-from numerary.types import CachingProtocolMeta  # type: ignore[attr-defined]
+if sys.version_info > (3, 7):
+    from numerary.protocol import CachingProtocolMeta
+else:
+    from numerary.types import CachingProtocolMeta  # type: ignore[attr-defined]
```
Comment (flaeppe) [inline src/phantom/_utils/types.py:7]: Gonna have to double check that this would in fact fix the caching issue

**Comment by antonagestam** [comment]
@flaeppe Thanks, I fixed the issue with pre-commit, it should pass now after a sync with main.

File: src/phantom/_utils/types.py (hunk 1 above)
Comment (antonagestam) [reply src/phantom/_utils/types.py:7]: If it doesn't, perhaps worth trying with `if sys.version_info > (3, 7) or TYPE_CHECKING:`?

File: src/phantom/_utils/types.py [hunk 2]
```diff
@@ -1,6 +1,10 @@
+import sys
 from typing import TypeVar
 
-from numerary.types import CachingProtocolMeta  # type: ignore[attr-defined]
+if sys.version_info > (3, 7):
```
Comment (antonagestam) [inline src/phantom/_utils/types.py:4]:
```suggestion
if sys.version_info >= (3, 8):
```

I'm guessing we're still failing because 3.7.15 > 3.7?

Comment (flaeppe) [reply src/phantom/_utils/types.py:4]: Yep, just realised, haha

File: src/phantom/_utils/types.py (hunk 1 above)
Comment (flaeppe) [reply src/phantom/_utils/types.py]: That's an interesting idea!

Comment (flaeppe) [reply src/phantom/_utils/types.py]: Oh, seems it did not in fact work: https://github.com/antonagestam/phantom-types/actions/runs/3483387875/jobs/5826831019

Comment (flaeppe) [reply src/phantom/_utils/types.py]: > If it doesn't, perhaps worth trying with `if sys.version_info > (3, 7) or TYPE_CHECKING:`?

Hm, but I think by doing this, it'd eliminate the possibility of type checking successfully with < 3.8 ?

**Comment by antonagestam** [comment]
I think type tests need updating for latest mypy, [attempting to hold it back for now.](https://github.com/antonagestam/phantom-types/pull/246)

File: src/phantom/_utils/types.py (hunk 1 above)
Comment (antonagestam) [reply src/phantom/_utils/types.py]: Hmm, I thought the caching issue only occurs with numerary>=0.4.4? I'll gladly admit I haven't fully understood yet what's going on there though 🫣

Comment (flaeppe) [reply src/phantom/_utils/types.py]: I can't say I've bisected or so to figure out if it works properly with older versions or what triggers what more exactly.

It's a bit difficult due to cached mypy runs etc.

At least it seems the ignored error is cause, due to what type mypy decides to write to cache(`Any`) and that with latest pypi version of `numerary`

Comment (antonagestam) [reply src/phantom/_utils/types.py]: Gotcha, I think this fix is good anyways so we can move forward it. I do need to fix sphinx before I can make a release though ...

**Pull request merged**